### PR TITLE
fix: internal backup server url

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?
Change default internal server server configuration to use full URL to the backup server. This allows to access it from a different namespace. Such behaviour is needed for operator running in all namespaces mode.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-1497

### How to test this PR?
Try to create a backup using internal backup server in all namespaces operator mode.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: controllers/checlusterbackup/internal_backup_server.go [hunk 1]
```diff
@@ -264,21 +264,27 @@ func getInternalBackupServerConfigurationSpec(bctx *BackupContext) *chev1.CheBac
...
 		Port:                        backupServerPort,
 		RepositoryPath:              "che",
 		RepositoryPasswordSecretRef: BackupServerRepoPasswordSecretName,
 	}
 }
 
+// getInternalRestBackupServerUrl returns full service URL to be able to access it even from a different namespace.
+// This is needed for all namespaces operator mode.
+func getInternalRestBackupServerUrl(bctx *BackupContext) string {
+	return fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", backupServerServiceName, bctx.namespace, backupServerPort)
```
Comment (tolusha) [inline controllers/checlusterbackup/internal_backup_server.go:285]:
```suggestion
	return fmt.Sprintf("http://%s.%s.svc:%d", backupServerServiceName, bctx.namespace, backupServerPort)
```

Comment (tolusha) [reply controllers/checlusterbackup/internal_backup_server.go:285]: If internal communication is disabled we have to use public urls.

Comment (mmorhun) [reply controllers/checlusterbackup/internal_backup_server.go:285]: Ok, thank you. Will rework it.

File: controllers/checlusterbackup/internal_backup_server.go [hunk 2]
```diff
@@ -264,16 +301,26 @@ func getInternalBackupServerConfigurationSpec(bctx *BackupContext) *chev1.CheBac
...
 		},
 		Spec: chev1.CheBackupServerConfigurationSpec{
-			Rest: getExpectedInternalRestServerConfiguration(),
+			Rest: getExpectedInternalRestServerConfiguration(bctx),
 		},
 	}
 }
 
-func getExpectedInternalRestServerConfiguration() *chev1.RestServerConfig {
+func getExpectedInternalRestServerConfiguration(bctx *BackupContext) *chev1.RestServerConfig {
```
Comment (tolusha) [inline controllers/checlusterbackup/internal_backup_server.go:309]: Pls use internal url always

Comment (mmorhun) [reply controllers/checlusterbackup/internal_backup_server.go:309]: Ok, but it means that internal backup server will not be accessible in all namespaces operator mode if internal (across namespaces) communication is not allowed.

File: pkg/backup_servers/backup_servers.go [hunk 3]
```diff
@@ -29,7 +29,8 @@ type BackupServer interface {
 	// InitRepository creates backup repository on the backup server side.
 	InitRepository() (bool, error)
 
-	// IsRepositoryExist check whether the repository alredy innitialized.
+	// IsRepositoryExist check whether the repository alredy initialized.
```
Comment (tolusha) [inline pkg/backup_servers/backup_servers.go:32]:
```suggestion
	// IsRepositoryExist check whether the repository already initialized.
```

**Review by tolusha — APPROVED** [review]

**Pull request merged**